apps/labs/posts/sparse-array-ecosystem.md

```diff
@@ -0,0 +1,118 @@
+
+All formats have a number of read/write characteristics. For reading, they can be random access or sequential access, similarly with adding data to a sparse array. Whether they are ordered sequentially in memory is important, as this improves performance and compactness of the format. We list these characteristics below together with which library supports a format. We have also included a row for N-dimensional capabilities.
+
+|                   | **Random Read** | **Random Write** | **Cache Optimized** | **`sparse`** | **`scipy.sparse`** | **`cupyx.sparse`** | **`torch.sparse`** |
+| ----------------- | --------------- | ---------------- | ------------------- | ------------ | ------------------ | ------------------ | ------------------ |
+| **COO**           | 🚫              | 🚫               | ✅                  | 🚧           | 🚧                 | 🚧                 | ✅                 |
+| **DOK**           | ✅              | ✅               | 🚫                  | 🚧           | 🚧                 | 🚧                 | 🚫                 |
+| **CSR/CSC**       | 🚫              | 🚫               | ✅                  | 🚧           | ✅                 | ✅                 | 🚧                 |
+| **LIL**           | 🚫              | ✅               | 🚫                  | 🚫           | 🚧                 | 🚫                 | 🚫                 |
+| **DIA**           | ✅              | 🚫               | ✅                  | 🚫           | 🚧                 | 🚧                 | 🚫                 |
+| **BSR**           | ✅<sup>1</sup>  | ✅<sup>1</sup>   | ✅                  | 🚫           | 🚧                 | 🚫                 | 🚧                 |
+| **N-dimensional** |                 |                  |                     | ✅           | 🚫                 | 🚫                 | ✅                 |
+
+<sup>1</sup> Within same block if it exists <br />
+✅ Available, with optimized loops for atomic operations <br />
+🚧 Present, with intermediate conversions or sub-optimal iterations <br />
+🚫 Unavailable
+
+### Array Creation API
+
+Here, we will compare how to create sparse arrays with the different libraries, as well as any kinks and notable pain points.
+
+#### `sparse`
+
+`sparse` provides two main options for constructing arrays:
+
+1. From an existing array with [`sparse.asarray`](https://sparse.pydata.org/en/stable/generated/sparse.asarray.html).
+2. By creating a [`sparse.DOK`](https://sparse.pydata.org/en/stable/generated/sparse.DOK.html) instance, populating it, then using [`sparse.DOK.asformat`](https://sparse.pydata.org/en/stable/generated/sparse.DOK.asformat.html).
+3. By specifying the constituent data or arrays.
+
+One notable pain point for `sparse` is its performance: It's notably slower than equivalent SciPy operations.
```

Comment:
Things published on the web can have a long shelf life. So I'd be careful about speaking in definite terms, especially if, as we may hope, the landscape shifts in the next couple of years.

I think you can flesh out some content here by showing a plot comparing perf results for array creation across the different libraries.

Reply:
I've reworded this and added the fact that work is ongoing in https://github.com/Quansight/Quansight-website/pull/837/commits/3ee3120e76230d686f4a878658b4fe468e98267a.